### PR TITLE
:seedling: TK-2103 : Added Test Cases for Preflight Script

### PR DESCRIPTION
**Pull request by pallav-trilio**

### CODE REVIEW

File: tests/preflight/shunit2 [hunk 1]
```diff
@@ -0,0 +1,1048 @@
+#! /bin/sh
+# $Id: shunit2 335 2011-05-01 20:10:33Z kate.ward@forestent.com $
```
Comment (msurudwad) [inline tests/preflight/shunit2:2]: remove author and license info

**Review by msurudwad — CHANGES_REQUESTED** [review]
Update Run Preflight Integration Tests step of preflight job to run shell test-suite.

**Comment by msurudwad** [comment]
> Update Run Preflight Integration Tests step of preflight job to run shell test-suite.

Address this review comment in this PR as well… this way we'll know if test-cases are passing or not. 

Make sure shell lint passes and all jobs passes too in order to approve this PR.

**Review by msurudwad — CHANGES_REQUESTED** [review]
1. Randomize `dnsutils` pod name as well, in preflight script. Also, check for other resources which are using hardcoded name.
2. How are you taking care of cleaning up of residual resources created from tests, if test-cases fail?

File: tests/preflight/preflight_test.sh [hunk 2]
```diff
@@ -76,11 +76,15 @@ testVolumeSnapshot() {
 }
 
 cleanAll() {
-
   cleanup
   assertEquals 0 $?
 }
 
+set -o errexit
```
Comment (msurudwad) [inline tests/preflight/preflight_test.sh:83]: set this at start of script

File: tests/preflight/preflight_test.sh [hunk 3]
```diff
@@ -76,11 +76,15 @@ testVolumeSnapshot() {
...
+set -o pipefail
```
Comment (msurudwad) [inline tests/preflight/preflight_test.sh:84]: set this at start of script

**Review by msurudwad — CHANGES_REQUESTED** [review]
Resolve review comments… otherwise LGTM

File: tests/preflight/preflight_test.sh (hunk 2 above)
Comment (pallav-trilio) [reply tests/preflight/preflight_test.sh]: resolved

File: tests/preflight/preflight_test.sh (hunk 3 above)
Comment (pallav-trilio) [reply tests/preflight/preflight_test.sh]: resolved

**Review by madhurtrilio — APPROVED** [review]

**Pull request merged**